nucleus/core/bootstrap/src/main/java/com/sun/enterprise/glassfish/bootstrap/GlassfishBootstrapClassLoader.java

```diff
@@ -20,24 +20,27 @@
 import java.io.IOException;
 import java.net.MalformedURLException;
 import java.net.URL;
+import java.net.URLClassLoader;
 import java.nio.file.Paths;
 import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.List;
+import java.util.stream.Collectors;
 
 
 /**
  * This classloader is located between OSGI classloader and system JDK extensions classloader,
  * so it can provide early initialized libraries (before OSGI startup, so even OSGI can use them).
  * <p>
- * Examples
+ * List:
  * <ul>
  * <li>GlassFish Java Util Logging Extension
  * <li>Grizzly NPN API
  * </ul>
  *
  * @author David Matejcek
  */
-public class GlassfishBootstrapClassLoader extends GlassfishUrlClassLoader {
+public class GlassfishBootstrapClassLoader extends URLClassLoader {
```

Comment:
Could this extend `o.g.c.util.GlassfishUrlClassLoader` to share `toString`?

Reply:
No, because this CL is specific (bootstrap) - it must not have dependencies from it's children classloaders.

Reply:
I don't understand, so checking it with
- https://github.com/eclipse-ee4j/glassfish/pull/24227

Can you predict what will fail?

Reply:
Simply don't do that. You can't even add logging to the GFurlcl, because it would go against initialization. See GlassFishMain.

Reply:
Could `o.g.c.util.GlassfishUrlClassLoader` have `public static String toString(URLClassLoader)` - to be used by both `GlassfishUrlClassLoader.toString()` and `GlassfishBootstrapClassLoader.toString()` to keep the format the same?

Reply:
That makes again another common dependency for both while we simply want to keep them isolated. The bootstrap part is at the beginning a "semiisolated" JVM starting another. That's the reason of all that reflection in GlassfishMain.
And it still causes some pain on some systems. I don't like code duplication too, but here it has some reason and this is really short piece of code.
Maybe it would be possible to make that separation more clear somehow, on the other hand ... I see 100+ high priority security issues in Snyk ... I also created several issues for WebappCL, seen in logs, that's my direction here.